.buildkite/pipeline.elastic-agent-package.yml

```diff
@@ -126,3 +127,38 @@ steps:
         export DRA_VERSION
       fi
       .buildkite/scripts/steps/dra-publish.sh
+
+  - label: "Publishing via BK API for Independent Agent Release"
+    if: build.env("BUILDKITE_TRIGGERED_FROM_BUILD_PIPELINE_SLUG") == "independent-agent-release-staging"
+    key: bk-api-publish-independent-agent
+    depends_on: package
+    agents:
+      provider: "gcp"
+    env:
+      DRA_PROJECT_ID: "elastic-agent-package"
+      DRA_PROJECT_ARTIFACT_ID: "agent-package"
+    command: |
+      echo "+++ Restoring Artifacts"
+      buildkite-agent artifact download "build/**/*" .
+      echo "+++ Changing permissions for the BK API commands"
+      sudo chown -R :1000 build/distributions/
+      echo "--- File listing"
+      ls -alR build
+      echo "--- Copy workaround for ironbank container filename"
+      .buildkite/scripts/steps/ironbank-cp-workaround.sh
+      echo "--- File listing after workaround"
+      ls -alR build
+      echo "+++ Checking artifact validity with release-manager collect dry run"
+      DRA_DRY_RUN="--dry-run"
+      export DRA_DRY_RUN
+      .buildkite/scripts/steps/dra-publish.sh
+      echo "+++ Call buildkite-agent upload to save artifacts for job that triggered us"
+      buildkite-agent artifact upload "build/distributions/*"
+      echo "+++ Set job metadata if TRIGGER_JOB_ID is properly set"
+      if [[ -z "${TRIGGER_JOB_ID}" ]]; then
+        echo "TRIGGER_JOB_ID is not set, so not setting metadata"
+      else
+        echo "Setting metadata for job that trigger this one"
+        buildkite-agent meta-data set "triggered_build_id" "$BUILDKITE_BUILD_ID" --job $TRIGGER_JOB_ID
```

Comment:
How are we going to use this information later, why do we want to save them?

Reply:
By setting the `triggered_build_id` metadata field in the build that triggered this one, it passes is the Build ID of this `elastic-agent-package` build to the triggering build.  

The triggering build then uses this in a command to pull down the artifacts from this specific build.

Here's the command in the `unified-release` repo where we reference this: https://github.com/elastic/unified-release/blob/main/catalog/step-get-agent-bk-artifacts.go#L15

It's currently one of the easiest ways for a calling build to pull artifacts back from a triggered build in Buildkite.  Other ways involve obtaining a Buildkite API token that then has to be managed in vault, etc.  

Here's the example we took this from: https://forum.buildkite.community/t/how-to-download-artifacts-back-from-triggered-pipeline/3480/2

Reply:
oh ok, it make sense then. 

Can you put some comment about this reference, because if someone is going to touch this line and break it, the time to debug would be huge, especially if they don't have the full context.